mpxe/integration_tests/test_babydriver.py

```diff
@@ -0,0 +1,49 @@
+import time
+import unittest
+from mpxe.integration_tests import int_test
+from mpxe.sims import sim
+import sys
+sys.path.append('projects/baby_driver/scripts')
+import adc_read
+import can_util
+import gpio_get
+import gpio_interrupts
+import spi_exchange
+import i2c_write
+import gpio_set
+import repl_setup
+
+class TestBabyDriver(int_test.IntTest):
+    def setUp(self):
+        super().setUp()
+        self.babydriver = self.manager.start('baby_driver')
+        can_util.default_channel = "vcan0"
+        repl_setup.call_gpio_it()
+
+    def test_babydriver_adc_read(self):
+        time.sleep(1)
+        adc_read.adc_read(port='A', pin=6, raw=True)
+    def test_gpio_get(self):
+        time.sleep(1)
+        gpio_get.gpio_get('A', 6)
+    def test_gpio_set(self):
+        time.sleep(1)
+        self.babydriver.sim.get_gpio(proj=self.babydriver, port='A', pin=5)
+        time.sleep(1)
+        gpio_set.gpio_set('A', 5, True)
+        time.sleep(1)
+        self.babydriver.sim.get_gpio(proj=self.babydriver, port='A', pin=5)
+    def test_gpio_interrupts(self):
+        time.sleep(1)
+        gpio_interrupts.register_gpio_interrupt(port='A', pin=3)
+        time.sleep(1)
+        gpio_interrupts.unregister_gpio_interrupt(port='A', pin=3)
```

Comment:
To test gpio interrupts even better, you should simulate an interrupt:
* Pass a callback to `register_gpio_interrupt`, like `callback=interrupt_callback`. It can be defined inside `test_gpio_interrupts` and it has to take a single parameter `info`, but it doesn't have to use it. It can just set some property on `self` to True so you know that it's called (set that property to False out of the function so it's defined).
* After registering the interrupt, simulate the interrupt occurring. By default the interrupts are set on both rising and falling edges, so use `self.babydriver.sim.set_gpio` to set the pin to low and then high so you know the interrupt should occur.
* Then assert that the interrupt callback was called by asserting that the property set in the callback function is True.
This way you test all the use cases of gpio interrupts.

Reply:
I have a few questions about this one:

1. Should I define an inner function inside `test_gpio_interrupts`  called `interrupt_callback`  that takes a parameter `info` which is set to True in the body of the function, and then pass this function to `register_gpio_interrupts` as `callback=interrupt_callback` ? 
2.  For the paramater `info`, Do I declare and set it to False in `test_gpio_interrupts` and then call it in the inner function `interrupt_callback` and set it to True in its body, and finally assert that `info is True`? so that I can test if the callback actually happened.

Reply:
Please review the way I implemented it. 

The assert still fails but I am not quite sure if the way I coded is similar to the way you had in mind.

Reply:
The way you've implemented it is generally correct just with a few errors, I'll review it in a proper review in a moment.